v3enginetest/main.go

```diff
@@ -32,8 +33,11 @@ func check(err error) {
 }
 
 func checkWatchResp(resp []clientv3.WatchResponse) {
-	if len(resp) != 3 {
-		panic(fmt.Errorf("not the correct amount of responses returned from the watch channel"))
+	// This number isn't deterministic, because it can't be known
+	// how many watch events are captured in terms of TTLs and
+	// when the engine finishes shutting down.
+	if len(resp) < 3 {
```

Comment:
The original logic worked, because there are three watchers, which each received a single event. Now a much higher number events is captured with the exact number being non-deterministic.